### ISSUE TITLE
SEACAS configure failures in ATDM Trilinos 'ats2', 'cts1', 'waterman', and 'van1-tx2' builds stating 2020-05-23

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#7420]
CC: @trilinos/seacas , @kddevin (Trilinos Data Services, Product Lead), @gsjaardema 

<Checklist>
<???: Add label "client: ATDM">
<???: Add label "ATDM Sev: Blocker" (by default but could be other "ATDM Sev: XXX")>
<???: Add label "type: bug"?>
<???: Add label "impacting: tests"?>
<???: Add label for affected packages (e.g. "pkg: MueLu", "pkg: Tpetra", "pkg: Kokkos", etc.)>
<???: Add label "PA: ???Project Area???" (e.g. "PA: Linear Solvers", "PA: Data Services")>
<???: Add milestone "Initial cleanup of new ATDM ..." or "Keep promoted ATDM ...">
<???: Once GitHub Issue is created, add entries for tests to TrilinosATDMStatus/*.csv files>

## Next Action Status

<status-and-or-first-action>

## Description

As shown in [this query](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2020-05-23&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=configureerrors&compare2=43&value2=0) there were configure errors in all of the builds on the ATDM systems  'ats2', 'cts1', 'waterman', and 'stira' in the builds:

| Site | Build Name | C Err | C Warn | B Err | B Warn | Not Run | Fail | Pass | Proc Time | Start Time | Labels |
| -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
vortex | Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg | 1376c3 | 1 | 5 |   |   | 10 hours ago | (28 labels)
vortex | Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg_cuda-aware-mpi |   | 1 | 5 |   |   | 10 hours ago | (28 labels)
vortex | Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt | 1376c3 | 1 | 5 |   |   | 10 hours ago | (28 labels)
vortex | Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt_cuda-aware-mpi |   | 1 | 5 |   |   | 10 hours ago | (28 labels)
vortex | Trilinos-atdm-ats2-gnu-7.3.1-spmpi-2019.06.24_serial_static_dbg | 1376c3 | 1 | 4 |   |   | 10 hours ago | (28 labels)
vortex | Trilinos-atdm-ats2-gnu-7.3.1-spmpi-2019.06.24_serial_static_opt | 1376c3 | 1 | 4 |   |   | 10 hours ago | (28 labels)
eclipse | Trilinos-atdm-cts1-intel-19.0.5_openmpi-4.0.1_openmp_static_dbg | 1376c3 | 1 | 4 | 0 | 0 | 10 hours ago | (28 labels)
eclipse | Trilinos-atdm-cts1-intel-19.0.5_openmpi-4.0.1_openmp_static_opt | 1376c3 | 1 | 4 | 0 | 0 | 10 hours ago | (28 labels)
waterman | Trilinos-atdm-waterman-cuda-9.2-debug | 1376c3 | 1 | 8 |   |   | 10 hours ago | (28 labels)
waterman | Trilinos-atdm-waterman-cuda-9.2-opt | 1376c3 | 1 | 8 |   |   | 10 hours ago | (28 labels)
waterman | Trilinos-atdm-waterman-cuda-9.2-rdc-release-debug | 1376c3 | 1 | 8 |   |   | 10 hours ago | (28 labels)
waterman | Trilinos-atdm-waterman-cuda-9.2-release-debug | 1376c3 | 1 | 8 |   |   | 10 hours ago | (28 labels)
waterman | Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt | 1376c3 | 1 | 8 |   |   | 10 hours ago | (28 labels)
waterman | Trilinos-atdm-waterman_cuda-9.2_shared_opt | 1376c3 | 1 | 8 |   |   | 10 hours ago | (28 labels)

started failing on testing day 2020-05-23.

These seem to all show the configure error:

```
CMake Error at /vscratch1/jenkins/vortex-slave/workspace/Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg/Trilinos/cmake/tribits/core/package_arch/TribitsAddAdvancedTest.cmake:1210 (MESSAGE):
  Must have EXEC, CMND, or COPY_FILES_TO_TEST_DIR for TEST_2
Call Stack (most recent call first):
  packages/seacas/libraries/ioss/src/main/CMakeLists.txt:297 (TRIBITS_ADD_ADVANCED_TEST)
```

## Current Status on CDash

The status of these tests/builds for the current testing day can be found at:

* [ATDM Trilinos builds for current testing day](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-)

## Steps to Reproduce

One should be able to reproduce this failure on the machine <supported-atdm-machine> as described in:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

More specifically, the commands given for the system 'cts1' (machines 'eclipse', 'attaway', etc.) are provided at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#tlcc-2-and-cts-1

The exact commands to reproduce this issue on a 'cts1' machine should be:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh \
    Trilinos-atdm-cts1-intel-19.0.5_openmpi-4.0.1_openmp_static_op

$ cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_SEACAS=ON \
 $TRILINOS_DIR
```

**Comment by bartlettroscoe** [comment]
@gsjaardema,  from looking at the commits pulled on 2020-05-23 shown [here](https://testing-dev.sandia.gov/cdash/viewNotes.php?buildid=5473677#!#note1), it looks likely that this error was triggered by merged PR #7408.

**Comment by bartlettroscoe** [comment]
Looking at that CMake code in:

https://github.com/trilinos/Trilinos/blob/develop/packages/seacas/libraries/ioss/src/main/CMakeLists.txt#L304

which shows:

```
TRIBITS_ADD_ADVANCED_TEST(exodus_to_unstructured_cgns_to_exodus
   TEST_0 EXEC io_shell ARGS ${DECOMP_ARG} ${CMAKE_CURRENT_SOURCE_DIR}/test/8-block.g 8-block.cgns
     NOEXEPREFIX NOEXESUFFIX
     NUM_MPI_PROCS ${NPROCS}
   TEST_1 EXEC io_shell ARGS ${JOIN_ARG} 8-block.cgns 8-block.g
     NOEXEPREFIX NOEXESUFFIX
     NUM_MPI_PROCS ${NPROCS}
   TEST_2 CMND ${SEACAS_EXODIFF_BINARY} ARGS -stat -pedantic -map -dumpmap ${CMAKE_CURRENT_SOURCE_DIR}/test/8-block.g 8-block.g
  COMM mpi serial
  XHOSTTYPE Windows
  )
```

it looks like the variable `SEACAS_EXODIFF_BINARY` is empty.

Any idea what is going on here?

**Comment by bartlettroscoe** [comment]
The cache shown [here](https://testing-dev.sandia.gov/cdash/viewNotes.php?buildid=5473677#!#note7) shows:

```
SEACAS_EXODIFF_BINARY:FILEPATH=SEACAS_EXODIFF_BINARY-NOTFOUND
```

Don't understand why this test is even getting enabled then.

**Comment by bartlettroscoe** [comment]
This has been fixed for a long time (see PR #7421).

**Comment by bartlettroscoe** [comment]
Not sure why this has not been closed yet but these errors were addressed a long time ago (see #7421).
